sdk/storage/azure-storage-common/src/storage_switch_to_secondary_policy.cpp

```diff
@@ -13,10 +13,7 @@ namespace Azure { namespace Storage { namespace _internal {
       const Azure::Core::Context& ctx) const
   {
     std::shared_ptr<bool> replicaStatus;
-    if (ctx.HasKey(SecondaryHostReplicaStatusKey))
-    {
-      replicaStatus = ctx.GetValue<std::shared_ptr<bool>>(SecondaryHostReplicaStatusKey);
-    }
+    ctx.TryGetValue<std::shared_ptr<bool>>(SecondaryHostReplicaStatusKey, replicaStatus);
```

Comment:
The way it is used, let's guarantee (in the docs) that the value will be untouched if the key was not found.

Reply:
This also begs the question if we use non-`Try` version of this method at all.